rust/private/rustc.bzl

```diff
@@ -35,179 +35,114 @@ CrateInfo = provider(
     },
 )
 
+DepInfo = provider(
+    fields = {
+        "direct_crates": "depset[CrateInfo]",
+        "indirect_crates": "depset[CrateInfo]",
+        "transitive_crates": "depset[CrateInfo]",
+        "transitive_dylibs": "depset[File]",
+        "transitive_staticlibs": "depset[File]",
+        "transitive_libs": "List[File]: All transitive dependencies, not filtered by type.",
+    },
+)
+
 def _get_rustc_env(ctx):
     version = ctx.attr.version if hasattr(ctx.attr, "version") else "0.0.0"
     major, minor, patch = version.split(".", 2)
     if "-" in patch:
         patch, pre = patch.split("-", 1)
     else:
         pre = ""
-    return [
-        "CARGO_PKG_VERSION=" + version,
-        "CARGO_PKG_VERSION_MAJOR=" + major,
-        "CARGO_PKG_VERSION_MINOR=" + minor,
-        "CARGO_PKG_VERSION_PATCH=" + patch,
-        "CARGO_PKG_VERSION_PRE=" + pre,
-        "CARGO_PKG_AUTHORS=",
-        "CARGO_PKG_NAME=" + ctx.label.name,
-        "CARGO_PKG_DESCRIPTION=",
-        "CARGO_PKG_HOMEPAGE=",
-    ]
+    return {
+        "CARGO_PKG_VERSION": version,
+        "CARGO_PKG_VERSION_MAJOR": major,
+        "CARGO_PKG_VERSION_MINOR": minor,
+        "CARGO_PKG_VERSION_PATCH": patch,
+        "CARGO_PKG_VERSION_PRE": pre,
+        "CARGO_PKG_AUTHORS": "",
+        "CARGO_PKG_NAME": ctx.label.name,
+        "CARGO_PKG_DESCRIPTION": "",
+        "CARGO_PKG_HOMEPAGE": "",
+    }
 
 def _get_compilation_mode_opts(ctx, toolchain):
     comp_mode = ctx.var["COMPILATION_MODE"]
     if not comp_mode in toolchain.compilation_mode_opts:
-        fail("Unrecognized compilation mode %s for toolchain." % comp_mode)
+        fail("Unrecognized compilation mode {} for toolchain.".format(comp_mode))
 
     return toolchain.compilation_mode_opts[comp_mode]
 
-def _get_lib_name(lib):
+# def get_lib_name(lib: File) -> str:
+def get_lib_name(lib):
     """Returns the name of a library artifact, eg. libabc.a -> abc"""
     libname, ext = lib.basename.split(".", 2)
-    if not libname.startswith("lib"):
-        fail("Expected {} to start with 'lib' prefix.".format(lib))
-    return libname[3:]
 
-def _symlink_dep_cmd(lib, deps_dir, in_runfiles):
-    """
-    Helper function to construct a command for symlinking a library into the
-    deps directory.
-    """
-    lib_path = lib.short_path if in_runfiles else lib.path
-    return (
-        "ln -sf " + relative_path(deps_dir, lib_path) + " " +
-        deps_dir + "/" + lib.basename + "\n"
-    )
+    if libname.startswith("lib"):
```

Comment:
I ran into this earlier with the llvm dylibs

Reply:
I ran into this downstream of a `cc_import` of llvm libraries, it's apparently valid naming.

Reply:
Ack, resolving.